utility/types/codec.go

```diff
@@ -0,0 +1,54 @@
+package types
+
+import (
+	"github.com/pokt-network/pocket/shared/types"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+type Codec interface { // TODO (Team) move to shared. Possibly rename
```

Comment:
I can actuall start using this file already. I think I made something similar in P2P, let's definitely share this

Reply:
However, the one I've made is generic and does not precisely expect `proto.Message`, so further interface casting should be accounted for after Unmarshalling